File: greenshine_customization/customization/lead/lead.py
import frappe
from frappe.model.mapper import get_mapped_doc

@frappe.whitelist()
def make_survey_form(source_name, target_doc=None):
	target_doc = get_mapped_doc("Lead", source_name,
		{"Lead": {
			"doctype": "Survey Form",
			"field_map": {
				"name": "lead",
				"lead_name": "company_name",
				"company_name": "name_of_kitchen",
				"email_id": "email",
				"mobile_no": "mobile"
			}
		}}, target_doc)

	return target_doc

File: greenshine_customization/customization/lead/lead.js
frappe.ui.form.on('Lead', {
    refresh: function(frm) {
        if(!frm.doc.__islocal && frm.doc.survey_form){
            
        
            frm.add_custom_button(__("Survey Form"), function(){
                frappe.model.open_mapped_doc({
                    method: "greenshine_customization.customization.lead.lead.make_survey_form",
                    frm: cur_frm
                })
            }, __("Make"));
        }
        frm.add_custom_button(__("Survey Form"), function(){
            console.log(frm);
			debugger;
			frappe.set_route("List", "Survey Form",{"lead": frm.doc.name})
            
        }, __("View"));
        
        frm.add_custom_button(__("Quotation"), function(){
            //perform desired action such as routing to new form or fetching etc.
            console.log(frm);
			debugger;
			frappe.set_route("List", "Quotation",{"party_name": frm.doc.name})
            
        }, __("View"));
    }
});